src/utils.js

```diff
@@ -120,7 +120,7 @@ export const requestPromise = async (options, stats) => {
 
         if (statusCode === RATE_LIMIT_EXCEEDED_STATUS_CODE && stats) {
             // Make sure this doesn't fail when someone increases number of retries on anything.
-            if (typeof stats.rateLimitErrors[iteration] === 'number') stats.rateLimitErrors[iteration - 1]++;
+            if (typeof stats.rateLimitErrors[iteration] === 'number') stats.rateLimitErrors[iteration]++;
```

Comment:
Actually it should be `iteration-1`, `iteration` is `1` in the first loop

Reply:
Hmm you are right, then the other two index into the array are probably wrong. Since the condition checks `iteration`, but then sets `iteration-1`, same for the else branch where it sets `iteration` to 1

Reply:
So, how to fix this? Maybe @mnmkng can advice?

Reply:
There needs to be `-1` everywhere I believe. I did not have the `if {} else {}` wrapper there initially and then I figured I'll throw it in "just in case", but apparently I did not update all the indices.